app/controllers/orders_controller.rb

```diff
@@ -1,43 +1,43 @@
 class OrdersController < ApplicationController
+  before_action :validate_user
+
+  def validate_user
+    if current_user.nil?
+      flash[:login] = {color: 'red', message: "Please log in to check out"}
```

Comment:
Space inside { missing.<br>Space inside } missing.<br>Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.